src/main/java/com/nekomaster1000/infernalexp/entities/ShroomloinEntity.java

```diff
@@ -157,6 +161,7 @@ public void tick() {
 		if (this.isAlive()) {
 		    if (this.isConverting() && this.conversionTicks > 0) {
 		        this.conversionTicks--;
+                this.addPotionEffect(new EffectInstance(Effects.SLOWNESS, this.conversionTicks, 2));
```

Comment:
I'd modify the actual movement speed attribute of the entity, not give it an effect.